test/unittest/blas2/blas2_gemv_test.cpp

```diff
@@ -42,40 +42,41 @@ void run_test(const combination_t<scalar_t> combi) {
 
   const char *t_str = trans ? "t" : "n";
 
-  int lda = m * lda_mul;
-  int x = trans ? m : n;
-  int y = trans ? n : m;
+  int a_size = m * n * lda_mul;
+  int x_size = trans ? (1 + (m - 1) * incX) : (1 + (n - 1) * incX);
+  int y_size = trans ? (1 + (n - 1) * incY) : (1 + (m - 1) * incY);
 
   // Input matrix
-  std::vector<scalar_t> a_m(lda * n);
+  std::vector<scalar_t> a_m(a_size, 1.0);
   // Input Vector
-  std::vector<scalar_t> b_v(x * incX);
+  std::vector<scalar_t> x_v(x_size, 1.0);
   // output Vector
-  std::vector<scalar_t> c_v_gpu_result(y * incY, scalar_t(10.0));
+  std::vector<scalar_t> y_v_gpu_result(y_size, scalar_t(0.0));
```

Comment:
Why do you change that to 0? I think that this is intended on purpose because when using a value different than 0 you'd notice e.g if your algorithm writes a 0 in a location that shouldn't be written (it can happen in that kind of algorithms that read zeroes when out of range, if the check before writing is wrong).

Reply:
Actually `y_v_gpu_result` could be random and `y_v_cpu` constructed by copy.

Reply:
Reverted